### PR TITLE
Cont3xt Vue 3 Migration

### PR DESCRIPTION
**Pull request by TobySalusky**
* migrated Cont3xt to Vue 3, Vite, & Vuetify
* `cont3xt/common` contains any common UI components updated for Vue3/Vuetify, and some .js files that needed change as well (vueFilters.js, git.js)
* import path aliases:
    * @: cont3xt/vueapp/src/
    * @common: cont3xt/common/vueapp/
    * @real_common: common/vueapp/
* ui testing is not implemented (and the pre-existing tests would likely need to be heavily reworked, so I've deleted them for now); thus, removed cont3xt from "npm run test" for the time being.
* some package changes to root level package.json, but I tried to keep most in cont3xt/package.json
* let me know what I should do about the package lock files? I think all of mine wound up getting updated

Let me know what you think :)

### CODE REVIEW

File: cont3xt/vueapp/src/vue3-helpers.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export function useGetters (store) {
+  return Object.fromEntries(
+    Object.keys(store.getters)
+      .map(getterKey => [getterKey, computed(() => store.getters[getterKey])])
+  );
+}
+
+export function escapeSelectorId (id) {
+  const escapedSpecialChars = id.replaceAll(/[:.*>~,$| ]/g, '\\$&');
```
[the remark this answers is not in the document]
Comment (TobySalusky) [reply cont3xt/vueapp/src/vue3-helpers.js]: Good catch! 🙇

File: cont3xt/common/vueapp/Roles.vue [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    },
+    userCountMemberString (count, unknown) {
+      // the text shown on the member dropdowns
+      return `${unknown ? '?' : count} ${count === 1 ? 'user' : 'users'}`;
+    }
+  }
+};
+</script>
+
+<style scoped>
```
Comment (31453) [inline cont3xt/common/vueapp/Roles.vue:159]: remove this

File: cont3xt/vueapp/src/utils/TimeRangeInput.vue [hunk 3]
```diff
@@ -3,231 +3,231 @@ Copyright Yahoo Inc.
...
+});
+
+watch(doneLoading, (val) => { // snap to the last snapTo value if it exists
+  if (val && localStorage.getItem('snapTo')) {
+    snapTo(parseInt(localStorage.getItem('snapTo')));
+  }
+});
 </script>
 
 <style scoped>
```
Comment (31453) [inline cont3xt/vueapp/src/utils/TimeRangeInput.vue:233]: remove

**Comment by 31453** [comment]
Tags don't show up before the search

**Comment by 31453** [comment]
Create Link Group not showing modal

**Comment by 31453** [comment]
weird wrapping
Screenshot 2024-09-06 at 4 02 34 PM
 in view cards

**Pull request closed without merging**